### PR TITLE
sink: fix avro timezone

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- https://asktug.com/t/topic/69408/4

### What is changed and how it works?
- Restore timezone information in Avro encoder.

### Check List 

Tests 

 - Unit test
 - Integration test
Related changes

 - Need to cherry-pick to the release branch

### Release note

- Fix time zone lost in Avro output.

### CODE REVIEW

**Comment by amyangfei** [comment]
Is this PR ready to review?

**Comment by liuzix** [comment]
> Is this PR ready to review?

Yes

**Comment by amyangfei** [comment]
Whether other protocols have this problem, including maxwell, canal, etc. If so, we can file an issue and assign tasks to freshman

File: cdc/sink/codec/avro.go [hunk 1]
```diff
@@ -379,19 +392,39 @@ func columnToAvroNativeData(col *model.Column) (interface{}, string, error) {
...
+			return time.Time{}, string(fullType), nil
+		}
+
+		var actualTz *time.Location
+		if col.Type != mysql.TypeTimestamp {
+			var err error
+			actualTz, err = time.LoadLocation("UTC")
+			if err != nil {
+				return nil, "", cerror.ErrAvroEncodeFailed.Wrap(err)
+			}
```
Comment (amyangfei) [inline cdc/sink/codec/avro.go:411]:
```suggestion
			actualTz = time.UTC
```
This is more simple

File: cdc/sink/codec/avro.go [hunk 2]
```diff
@@ -379,19 +392,35 @@ func columnToAvroNativeData(col *model.Column) (interface{}, string, error) {
...
-		str := col.Value.(string)
-		t, err := time.Parse(types.DateFormat, str)
+		// Refer to `unflatten` in cdc/entry/codec.go for why this piece of code is like this.
 		const fullType = "long." + timestampMillis
+		str := col.Value.(string)
+
+		if (col.Type == mysql.TypeDate && str == zeroDateStr) ||
+			(col.Type != mysql.TypeDate && str == zeroTimeStr) {
+
+			return time.Time{}, string(fullType), nil
```
Comment (overvenus) [inline cdc/sink/codec/avro.go:402]:
```suggestion
			return nil, string(fullType), nil
```

Why not nil?

**Review by overvenus — COMMENTED** [review]
rest lgtm

File: cdc/sink/codec/avro.go (hunk 2 above)
Comment (liuzix) [reply cdc/sink/codec/avro.go:402]: `nil` does not have the same type as `time.Time{}`, which will cause problem in the caller of this function.

**Comment by liuzix** [comment]
@overvenus @leoppro PTAL

**Pull request merged**